model/map/src/main/java/org/keycloak/models/map/storage/chm/MapFieldPredicates.java

```diff
@@ -306,6 +306,33 @@ private static MapModelCriteriaBuilder<Object, MapGroupEntity, GroupModel> check
         return mcb.fieldCompare(Boolean.TRUE::equals, getter);
     }
 
+    private static MapModelCriteriaBuilder<Object, MapUserEntity, UserModel> usernameCaseInsensitive(MapModelCriteriaBuilder<Object, MapUserEntity, UserModel> mcb, Operator op, Object[] values) {
+        if (values == null || values.length != 1) {
+            throw new CriterionNotSupportedException(UserModel.SearchableFields.USERNAME_CASE_INSENSITIVE, op, "Invalid arguments, expected (username), got: " + Arrays.toString(values));
+        }
+
+        final Object ob = values[0];
+        if (!String.class.isAssignableFrom(ob.getClass())) {
+            throw new CriterionNotSupportedException(UserModel.SearchableFields.USERNAME_CASE_INSENSITIVE, op, "Invalid arguments, expected (String), got: " + Arrays.toString(values));
+        }
+
+        String value = (String) ob;
+        Function<MapUserEntity, ?> getter;
+
+        switch (op) {
```

Comment:
It's implemented this way in `Jpa*ModelCriteriaBuilder`s. There it saves time when some operator is not implemented and is tried to be used then it fails fast with clear message. On the other hand if there is a way how to implement it for more generically it may be worth changing in a way you're proposing.